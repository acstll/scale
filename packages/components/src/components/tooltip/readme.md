# t-tooltip



<!-- Auto Generated Below -->


## Properties

| Property    | Attribute   | Description | Type                                               | Default  |
| ----------- | ----------- | ----------- | -------------------------------------------------- | -------- |
| `delay`     | `delay`     |             | `number \| object`                                 | `0`      |
| `offset`    | `offset`    |             | `number \| string`                                 | `''`     |
| `placement` | `placement` |             | `"auto" \| "bottom" \| "left" \| "right" \| "top"` | `'auto'` |
| `size`      | `size`      |             | `string`                                           | `''`     |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
