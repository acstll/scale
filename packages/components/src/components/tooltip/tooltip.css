.tooltip {
  background: var(--tooltip-background, #343a40);
  color: var(--tooltip-color, #fff);
  display: inline-flex;
  padding: var(--tooltip-padding, 0.25em 0.4em);
  font-size: var(--tooltip-font-size, 90%);
  font-weight: var(--tooltip-font-weight, 700);
  line-height: var(--tooltip-line-height, 1);
  text-align: var(--tooltip-text-allign, center);
  white-space: var(--tooltip-white-space, nowrap);
  vertical-align: var(--tooltip-vertical-align, baseline);
  border-radius: var(--tooltip-border-radius, 0.25rem);
  position: relative;
}
.tooltip--placement-bottom:after {
  content: '';
  position: absolute;
  top: var(0);
  left: 50%;
  width: 0;
  height: 0;
  border: 8px solid transparent;
  border-bottom-color: var(--tooltip-background, #343a40);
  border-top: 0;
  margin-left: -8px;
  margin-top: -12px;
  transition: var(
    --tooltip--placement-bottom-transition,
    --tooltip--placement-bottom-border,
    --tooltip--placement-bottom-border-bottom-color,
    --tooltip--placement-bottom-margin-left,
    --tooltip--placement-bottom-margin-top
  );
}

.tooltip--placement-top:after {
  content: '';
  position: absolute;
  bottom: 0;
  left: 50%;
  width: 0;
  height: 0;
  border: 8px solid transparent;
  border-top-color: var(--tooltip-background, #343a40);
  border-bottom: 0;
  margin-left: -8px;
  margin-bottom: -8px;
  transition: var(
    --tooltip--placement-top-transition,
    --tooltip--placement-top-border,
    --tooltip--placement-top-border-top-color,
    --tooltip--placement-top-margin-left,
    --tooltip--placement-top-margin-bottom
  );
}

.tooltip--placement-left:after {
  content: '';
  position: absolute;
  right: 0;
  top: 50%;
  width: 0;
  height: 0;
  border: 8px solid transparent;
  border-left-color: var(--tooltip-background, #343a40);
  border-right: 0;
  margin-top: -8px;
  margin-right: -8px;
  transition: var(
    --tooltip--placement-left-transition,
    --tooltip--placement-left-border,
    --tooltip--placement-left-border-left-color,
    --tooltip--placement-left-margin-top,
    --tooltip--placement-left-margin-right
  );
}

.tooltip--placement-right:after {
  content: '';
  position: absolute;
  left: 0;
  top: 50%;
  width: 0;
  height: 0;
  border: 8px solid transparent;
  border-right-color: var(--tooltip-background, #343a40);
  border-left: 0;
  margin-top: -8px;
  margin-left: -8px;
  transition: var(
    --tooltip--placement-right-transition,
    --tooltip--placement-right-border,
    --tooltip--placement-right-border-right-color,
    --tooltip--placement-right-margin-top,
    --tooltip--placement-right-margin-left
  );
}
