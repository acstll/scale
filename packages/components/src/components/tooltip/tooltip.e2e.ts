import { newE2EPage } from '@stencil/core/testing';

describe('t-tooltip', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<t-tooltip>Tooltip</t-tooltip>');
    const element = await page.find('t-tooltip');
    expect(element).toHaveClass('hydrated');
  });
});
