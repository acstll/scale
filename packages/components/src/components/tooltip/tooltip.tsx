import { Component, Prop, h } from '@stencil/core';
import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';

@Component({
  tag: 't-tooltip',
  styleUrl: 'tooltip.css',
  shadow: true,
})
export class Tooltip {
  @Prop() public size?: string = '';
  @Prop() public delay?: number | object = 0;
  @Prop() public placement?: 'auto' | 'top' | 'bottom' | 'left' | 'right' =
    'auto';
  @Prop() public offset?: string | number = '';

  public render() {
    return (
      <span class={this.getCssClassMap()}>
        <slot />
      </span>
    );
  }

  private getCssClassMap(): CssClassMap {
    return classNames(
      'tooltip',
      this.size && `tooltip--size-${this.size}`,
      this.placement && `tooltip--placement-${this.placement}`
    );
  }
}
