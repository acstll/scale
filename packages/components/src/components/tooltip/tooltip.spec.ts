import { newSpecPage } from '@stencil/core/testing';
import { Tooltip } from './tooltip';

describe('Tooltip', () => {
  let element;
  beforeEach(async () => {
    element = new Tooltip();
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Tooltip],
      html: `<t-tooltip>Tooltip</t-tooltip>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle size css class', () => {
    element.size = 'default';
    expect(element.getCssClassMap()).toContain('tooltip--size-default');
  });

  it('should handle placement css class', () => {
    element.placement = 'default';
    expect(element.getCssClassMap()).toContain('tooltip--placement-default');
  });
});
