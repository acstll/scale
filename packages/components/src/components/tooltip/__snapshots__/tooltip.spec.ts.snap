// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip should match snapshot 1`] = `
<t-tooltip>
  <mock:shadow-root>
    <span class="tooltip tooltip--placement-auto">
      <slot></slot>
    </span>
  </mock:shadow-root>
  Tooltip
</t-tooltip>
`;
